lib/screens/transfer/send/send_confirmation/interactor/viewmodels/send_confirmation_bloc.dart

```diff
@@ -31,13 +31,25 @@ class SendConfirmationBloc extends Bloc<SendConfirmationEvent, SendConfirmationS
     // We can extend this initial validation logic in future using a switch case for any transaction type
     // for now it only validates a transfer
     if (state.isTransfer) {
-      final esoAction = state.transaction.actions.first;
-      final symbol = (esoAction.data?['quantity'] as String).split(' ').last;
-      final targetToken = TokenModel.allTokens.singleWhereOrNull((i) => i.symbol == symbol);
-      if (targetToken != null) {
-        final Result<BalanceModel> result = await GetAvailableBalanceUseCase().run(targetToken);
-        emit(InitialValidationStateMapper().mapResultToState(state, result));
-      }
+      final eosAction = state.transaction.actions.first;
```

Comment:
What does this code do?

Reply:
This is dealing with the case when you send a token (e.g. with a QR code) which is _not_ in your list of known TokenModels. Some of the transfer code wanted to refer to a TokenModel, and rather than refactor that so it didn't need a model, this section creates a transient TokenModel object just for the transaction.
It felt clumsy when I wrote it, and it feels worse now. But it has been working reliably.

Reply:
Interesting edge case, never thought of that, but perfectly valid. 

👍